File: CodeForge__BE/src/CodeForge.Core/Entities/Problem.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace CodeForge.Core.Entities
{
        [Table("CodingProblems")]
        public class Problem
        {
                [Key]
                [Column("ProblemID")]
                public Guid ProblemId { get; set; } = Guid.NewGuid();

                public Guid? LessonId { get; set; }

                [Required, MaxLength(200)]
                public string? Title { get; set; }

                [Required, MaxLength(200)]
                public string? Slug { get; set; }

                public string? Description { get; set; }

                [Required, MaxLength(20)]
                public string? Difficulty { get; set; } = "Dễ";

                [MaxLength(255)]
                public string? Tags { get; set; }

                [MaxLength(100)]
                public string? FunctionName { get; set; }

                [MaxLength(500)]
                public string? Parameters { get; set; }

                [MaxLength(100)]
                public string? ReturnType { get; set; }

                public string? Status { get; set; } = "NOT_STARTED";

                public int TimeLimit { get; set; } = 1000;

                public int MemoryLimit { get; set; } = 256;

                public string? Notes { get; set; }

                public string? Constraints { get; set; }

                public DateTime CreatedAt { get; set; } = DateTime.Now;

                public DateTime UpdatedAt { get; set; } = DateTime.Now;

                public bool IsDeleted { get; set; } = false;

                // ---------------- 🔗 Navigation Property ----------------
                [ForeignKey("LessonId")]
                public Lesson? Lesson { get; set; }

                public Problem() { }
        }
}

File: CodeForge__FE/src/features/practice/types/problem.ts


export interface ProblemTestCase {
  input: Variable[];
  expectedOutput: string;
  explain: string;
  isHidden: boolean;
}

export interface Problem {
  problemId : string ;
  title: string;
  slug: string;
  difficulty: "Dễ" | "Trung Bình" | "Khó";
  status: "NOT_STARTED" | "ATTEMPTED" | "SOLVED";
  description: string;
  tags: string;
  functionName: string;
  parameters: string;
  returnType: string;
  notes: string;
  constraints: string;
  timeLimit: number;
  memoryLimit: number;
  lessonId: string | null;
}

export interface Variable {
  name: string;
  type: string;
  value: string;
}

export interface TestCase {
  testCaseId : string ;
  input: Variable[];
  expectedOutput: string;
  explain: string;
  isHidden: boolean;
}

export interface Lesson {
  id: string;
  name: string;
}

export interface DifficultyOption {
  value: "Dễ" | "Trung Bình" | "Khó";
  label: string;
  color: string;
}

export interface TabItem {
  id: string;
  label: string;
  icon: React.ComponentType<{ size?: number }>;
}

export interface Notification {
  message: string;
  type: "success" | "error";
}

export interface Errors {
  [key: string]: string;
}